[ADD] estate: created new module

haahi-odoo opened the pull request:
Initializes the basic structure for the real estate module.
- Creates the module scaffolding (manifest, init, and base directories).
- Defines the initial 'estate.property' model.
- Adds base fields (columns) to the new model.

haahi-odoo committed:
[ADD] estate: new 'estate' module

Initializes the basic structure for the real estate module.
- Creates the module scaffolding (manifest, init, and base directories).
- Defines the initial 'estate.property' model.
- Adds base fields (columns) to the new model.

haahi-odoo committed:
[ADD] estate: new 'estate' module

Initializes the basic structure for the real estate module.
- Creates the module scaffolding (manifest, init, and base directories).
- Defines the initial 'estate.property' model.
- Adds base fields (columns) to the new model.

robodoo commented:
[![Pull request status dashboard](https://mergebot.odoo.com/odoo/tutorials/pull/1020.png)](https://mergebot.odoo.com/odoo/tutorials/pull/1020)

haahi-odoo committed:
[IMP] estate: Configure field defaults and UI menus

- Set default values for estate.property model fields.
- Apply additional field attributes (e.g., readonly, required).
- Register menu items to access the tree and form views.

haahi-odoo committed:
[IMP] estate : Added list & form view

- added  basic, list and form view
- also added  filter for new or offer received stage

haahi-odoo committed:
[FIX] estate : mismatch field name

- pincode feild name had spelling mistake in views/estate_property_views.xml

haahi-odoo committed:
[FIX] estate: indentation error

haahi-odoo committed:
[IMP] estate: Implement list, form, and search views with grouping functionality

-Developed and integrated list, form, and search views for the Real Estate module
-Including support for group-by operations.

bit-odoo commented on estate/__manifest__.py:
```diff
@@ -0,0 +1,15 @@
+    'application': True,
+    'installable': True,
+    'author': 'Odoo S.A.',
+    'category':'Tutorials',
```
```suggestion
    'category': 'Tutorials',
```

bit-odoo commented on estate/__manifest__.py:
```diff
@@ -0,0 +1,15 @@
+        'views/estate_menus.xml',
+    ],
+}
+
```
An empty line should be without whitespace.

bit-odoo commented on estate/models/estate_property.py:
```diff
@@ -0,0 +1,40 @@
+        copy=False,
+        default='new'
+    )
+    
```
An empty line should be without whitespace.

bit-odoo commented on estate/models/estate_property.py:
```diff
@@ -0,0 +1,40 @@
+    living_area = fields.Integer("living_area(sqm)")
+    facades = fields.Integer("Facades")
+    garage = fields.Boolean("Garage")
+    garden = fields.Boolean("Garden")
```
By default, Odoo generates a string. If your string is the same as the technical name, then no need to add.

bit-odoo commented on estate/views/estate_menus.xml:
```diff
@@ -0,0 +1,21 @@
+
+    <menuitem id="estate_advertisement_menu"
+              name="Advertisements"
+              parent="estate_root_menu"/>
```
```suggestion
    <menuitem id="estate_advertisement_menu"
            name="Advertisements"
            parent="estate_root_menu"/>
```

bit-odoo commented on estate/views/estate_menus.xml:
```diff
@@ -0,0 +1,21 @@
+    <menuitem id="estate_property_form_menu"
+              name="Form"
+              parent="estate_property_menu"
+              action="estate_property_form_action"/>
```
same as above.

bit-odoo commented on estate/views/estate_property_views.xml:
```diff
@@ -0,0 +1,90 @@
+
+<odoo>
+   
```
unnecessary diff.

bit-odoo commented on estate/views/estate_property_views.xml:
```diff
@@ -0,0 +1,90 @@
+
```
unnecessary diff.

bit-odoo commented on estate/views/estate_property_views.xml:
```diff
@@ -0,0 +1,90 @@
+                <filter name="group_by_postcode"
+                        string="Group by Postcode"
+                        context="{'group_by': 'postcode'}"/>
+
```
unnecessary diff.

bit-odoo commented on estate/views/estate_property_views.xml:
```diff
@@ -0,0 +1,90 @@
+            <search string="Search Properties">
+                <field name="name"/>
+                <field name="postcode"/>
+
```
unnecessary diff.

bit-odoo commented on estate/views/estate_property_views.xml:
```diff
@@ -0,0 +1,90 @@
+                <separator/>
+                <filter name="group_by_postcode"
+                        string="Group by Postcode"
+                        context="{'group_by': 'postcode'}"/>
```
proper indentation needed.

bit-odoo commented on estate/views/estate_property_views.xml:
```diff
@@ -0,0 +1,90 @@
+    </record>
+            
+
+</odoo>
```
There should be one empty line at the end of the file.

bit-odoo reviewed (commented):
Hello,
Good Job!
I have left some remarks.

please update your commit message and title. you can refer this documentation :- https://www.odoo.com/documentation/19.0/contributing/development/git_guidelines.html#commit-message-full-description

haahi-odoo committed:
[IMP] estate: Enhanced Data Models and Functionalities

-Introduced Property Type and Property Tag models.
-Added Buyer and Salesperson fields.
-Implemented the Property Offer model.

haahi-odoo committed:
[IMP] estate: added  computed fields,inverse fun & onchange

-Added computed fields with inverse functions and onchange methods,
-Improved real estate module with sold and cancel buttons to enhance the module
functionality.

haahi-odoo committed:
[FIX] estate: fixed depreciated sql contraint

-Implemented new sql constraint method.

haahi-odoo committed:
[FIX] estate: fixed depreciated sql contraint

-Implemented new sql constraint method.

haahi-odoo committed:
[IMP] estate: added Inline Views, widgets, list order, Attributes, & Options.

-Implemented inline views and widgets to enhance the user interface within the
Real Estate module.
-Introduced list-ordering functionality to improve the organization and sorting
of listings.
-Added attribute and option configurations to increase customization capabilities
and overall flexibility for users.

bit-odoo commented on estate/models/estate_property.py:
```diff
@@ -0,0 +1,123 @@
+from dateutil.relativedelta import relativedelta
+from odoo import models, fields, api
+from odoo.exceptions import UserError, ValidationError
+from odoo.tools.float_utils import float_compare, float_is_zero
```
```suggestion
from dateutil.relativedelta import relativedelta

from odoo import models, fields, api
from odoo.exceptions import UserError, ValidationError
from odoo.tools.float_utils import float_compare, float_is_zero
```

bit-odoo commented on estate/models/estate_property.py:
```diff
@@ -0,0 +1,123 @@
+                    record.garden_orientation = "west"
+                elif record.garden_area == 40:
+                    record.garden_orientation = "south"
+                else:
```
Is this necessary?

bit-odoo commented on estate/views/estate_property_views.xml:
```diff
@@ -0,0 +1,119 @@
+                            <field name="garage" />
+                            <field name="garden" />
+                            <field name="garden_area" invisible="not garden" />
+                            <field name="garden_orientation" invisible="not garden" />
```
This part should be in the description tab.

bit-odoo commented on estate/models/estate_property.py:
```diff
@@ -0,0 +1,123 @@
+    _check_selling_price = models.Constraint(
+        'CHECK(selling_price > 0)',
+        'The selling price must be positive.'
+    )
```
Constraints should be after the field declaration; you can take reference from the Odoo codebase.

bit-odoo commented on estate/models/estate_property_offer.py:
```diff
@@ -0,0 +1,51 @@
+    _check_offer_price = models.Constraint(
+        'CHECK(price > 0)',
+        'The price must be strictly positive.'
+    )
```
same and should be one empty line at the end of the file.

bit-odoo commented on estate/models/estate_property_type.py:
```diff
@@ -0,0 +1,16 @@
+    _check_type_name_unique = models.Constraint(
+        'UNIQUE(name)', 
+        'The name of the property type must be unique.'
+    )
```
Should be one empty line at the end of the file.

bit-odoo commented on estate/views/estate_property_offer_views.xml:
```diff
@@ -0,0 +1,38 @@
+                decoration-success="status == 'accepted'" editable="bottom">
+                <field name="price" />
+                <field name="partner_id" />
+                <field name="status" invisible="1" />
```
Why invisible in the list view? If it is invisible, then you can remove it from the list view.

bit-odoo reviewed (commented):
Hello!
Another round of review :)
I have made some comments; please have a look at them.

Please update your commit message `Bootstrap new 'estate' module` and PR title too.

haahi-odoo committed:
[IMP] estate: Added stat button, related fields & business logic updates

-Introduced a stat button on Property Type to display related offer count.
-Restricted property deletion to New or Cancelled states, set status
 to Offer Received on offer creation, and prevented creation of offers lower
 than existing ones.

bit-odoo commented on estate/models/estate_property.py:
```diff
@@ -0,0 +1,124 @@
+
+    expected_price = fields.Float("Expected Price", required=True)
+    selling_price = fields.Float("Selling Price", readonly=True)
+
```
Empty line not required, unnecessary diff.

bit-odoo commented on estate/models/estate_property.py:
```diff
@@ -0,0 +1,124 @@
+        "Availability Date",
+        default=lambda self: fields.Date.today() + relativedelta(months=3),
+    )
+
```
Empty line not required, unnecessary diff.

bit-odoo commented on estate/models/estate_property.py:
```diff
@@ -0,0 +1,124 @@
+    living_area = fields.Integer("Living Area(sqft)")
+    facades = fields.Integer()
+    garage = fields.Boolean()
+
```
Empty line not required, unnecessary diff.

bit-odoo commented on estate/models/estate_property.py:
```diff
@@ -0,0 +1,124 @@
+        ],
+        string="Garden Orientation",
+    )
+
```
Empty line not required, unnecessary diff.

bit-odoo commented on estate/models/estate_property.py:
```diff
@@ -0,0 +1,124 @@
+        copy=False,
+        default="new",
+    )
+
```
Empty line not required, unnecessary diff.

bit-odoo commented on estate/models/estate_property.py:
```diff
@@ -0,0 +1,124 @@
+    )
+
+    active = fields.Boolean(default=True)
+
```
Empty line not required, unnecessary diff.

bit-odoo commented on estate/models/estate_property.py:
```diff
@@ -0,0 +1,124 @@
+        'The expected price of a property must be strictly positive.',
+    )
+
+
```
Should be only one empty line. Please check odoo base code for reference.

bit-odoo commented on estate/models/estate_property.py:
```diff
@@ -0,0 +1,124 @@
+        "property_id",
+        string="Offers"
+    )
+
```
Empty line not required, unnecessary diff.

bit-odoo commented on estate/models/estate_property.py:
```diff
@@ -0,0 +1,124 @@
+    salesperson_id = fields.Many2one("res.users", string="Salesperson")
+
+    tag_ids = fields.Many2many("estate.property.tag", string="Tags")
+
```
Empty line not required, unnecessary diff.

bit-odoo commented on estate/models/estate_property.py:
```diff
@@ -0,0 +1,124 @@
+    property_type_id = fields.Many2one("estate.property.type", "Property Type")
+    buyer_id = fields.Many2one("res.partner", "Buyer", copy=False)
+    salesperson_id = fields.Many2one("res.users", string="Salesperson")
+
```
Empty line not required, unnecessary diff.

bit-odoo commented on estate/models/estate_property_offer.py:
```diff
@@ -0,0 +1,99 @@
+        > avoid search inside loop
+        > validate in bulk
+        > update property state
+        """
```
The comment should be important.

bit-odoo commented on estate/models/estate_property_offer.py:
```diff
@@ -0,0 +1,99 @@
+
+        return records
+
+    _check_offer_price = models.Constraint(
```
Constraints should be after the field declaration.

bit-odoo commented on estate/models/estate_property_type.py:
```diff
@@ -0,0 +1,23 @@
+        for record in self:
+            record.offer_count = len(record.offer_ids)
+
+    _check_type_name_unique = models.Constraint(
```
Same here.

bit-odoo reviewed (commented):
Hello,

As i suugested you in early comments that your runbot should be green.
I have to write the same comments that i already suggested you in earlier comments.
You have to adapt the changes. I don't need to write it for every line and file.

haahi-odoo committed:
[IMP] estate : Added invoicing for sold properties and Kanban views

-Added two invoice function during invoice creation for sold properties
-One for 6% commission for sales and another for fixed charges of 100.
-Added a Kanban view to improve property visualization in the Real Estate module.

haahi-odoo committed:
[IMP] estate: code formation as per odoo guidelines

-fixed indentation errors
-fixed minor bugs
-code done as per guidelines mentioned in documentation

haahi-odoo committed:
[IMP] awesome_owl: add counter, card components and todo lists

-Made  estate module user-friendly UI
-Added counter and card component and display in awesome owl module
-Added todo list in awesome_owl

bit-odoo commented:
Hello @haahi-odoo 

Please push with a proper commit message. I have multiple times suggested this.

<img width="1665" height="420" alt="haahi" src="https://github.com/user-attachments/assets/8a7dace4-b4a3-4ce2-9a2f-c3b38a3008f7" />

haahi-odoo committed:
[IMP] awesome_owl: added new layout, buttons and a dashboard item

-added a new layout
-added some buttons for quick navigation
-added a new dashboard item

bit-odoo commented:
@haahi-odoo, multiple times I informed you that make sure your runbot is built green. But I clearly see that you didn't take it seriously.

Why do both commits have the same commit message?

<img width="1674" height="313" alt="image" src="https://github.com/user-attachments/assets/8b53d78f-7ca4-4211-84eb-074771273c23" />

haahi-odoo committed:
[IMP] awesome_owl: added statistics, server calls and a service

-Added functionality to call the server
-Added some statistics
-Cached the network calls
-Created a new Service

haahi-odoo committed:
[IMP] awesome_dashboard: added item list and pie chart

-Added dashboard item which is number card and pie chart card
-Added lazy loading functionality for dashboard
-Made our dashboard generic
-Added functionality to add and remove dashboard items
